### ISSUE TITLE
Stuttery NPC animations

### ISSUE THREAD

**Issue by TomArrow** [kugelrund/Speed-Outcast#2]
Hi, I'm getting stuttery NPC animations with this client. In vanilla JK2 (official release) single player that's not the case. Any idea what I'm doing wrong or what might be causing it?

**Comment by TomArrow** [comment]
Hmm seems it works fine when I set sv_fps to the default of 20. Is this considered normal and desired then? Or still a bug worth fixing?

**Comment by kugelrund** [comment]
I am not aware of any difference in behavior regarding NPC animations between Speed-Outcast and vanilla. If you use the same `com_maxfps` and `sv_fps` in either, it should behave the same. If that is not the case, then I would consider it a bug. I would need a more elaborate description about the specific cases you observe that in then.

**Comment by TomArrow** [comment]
Yeah it might have been my mistake to assume that it's a speed outcast specific problem. Vanilla might behave the same at sv_fps 125. I looked at it more closely and it seems it's not the animations themselves as much as the movement of the enemies. It's like they move at a lower fps. Their animations are fluid, but their position updates are once every few frames. So for example when you see a running stormtrooper, he will look stuttery at sv_Fps 125. 

This might be the same behavior as vanilla at same sv_fps but since vanilla wasn't meant to be played at sv_fps 125 one could argue that it changes the behavior of the game and might warrant a fix. Honestly I guess it's not a big deal for speedruns but it does kinda ruin the immersion for a normal playthrough.

**Comment by kugelrund** [comment]
In that case this is a "won't fix" for me for now. I do not intent to deviate from vanilla behavior without good reason. For discussing about the rules of what fps to speedrun with you'll have to talk to the community, that's not really my business. For all I know, runs done on `sv_fps 20` are fine for the leaderboards though, so you can use that if you like.

**Comment by TomArrow** [comment]
Ah. Understandable, thanks for your feedback.

**Comment by stacksandsheaves** [comment]
I have a related question about this, not an open issue though because this may be on my end, but when I do `sv_fps 20` in the console to handle the stutter issue, it latches but upon exiting the program and restarting it, I find that sv_fps is 125 again. I'm not sure if I'm misunderstanding what it means when it asks for a restart or if there's an issue here.

**Comment by kugelrund** [comment]
The latching was introduced to avoid people changing it on the fly during runs which was not wished by the community. A latched value is not to be applied after a game restarts though (unless the latched value is written to the config), it is usually about subsystem restarts. In the case of `sv_fps`, the latched value will be applied after a `vid_restart` or e.g. on a map-change.
